pyomo/core/tests/transform/test_zero_sum_propagate.py

```diff
@@ -0,0 +1,73 @@
+"""Tests the zero sum propagation module."""
+import pyutilib.th as unittest
+from pyomo.environ import (ConcreteModel, Constraint, TransformationFactory,
+                           Var, NonNegativeReals, NonPositiveReals)
+
+__author__ = "Qi Chen <qichen at andrew.cmu.edu>"
+
+
+class TestZeroSumPropagate(unittest.TestCase):
+    """Tests zero sum propagation."""
+
+    def test_non_negative_propagate(self):
+        """Test zero sum propagation for non-negative vars."""
+        m = ConcreteModel()
+        m.v1 = Var(initialize=0)
+        m.v2 = Var(initialize=2, domain=NonNegativeReals)
+        m.v3 = Var(initialize=3, domain=NonNegativeReals)
+        m.v4 = Var(initialize=4, domain=NonNegativeReals)
+        m.c1 = Constraint(expr=m.v1 == m.v2 + m.v3 + m.v4)
```

Comment:
For the sake of completeness, can you also test `m.v2 + m.v3 + m.v4 == m.v1`?  Mathematically it is the same, but results in opposite signs for the linear coefficients.